src/main/scala/io/shiftleft/js2cpg/cpg/passes/AstCreationPass.scala

```diff
@@ -47,36 +45,32 @@ class AstCreationPass(srcDir: File,
         Some((parseResult, preAnalyze(parseResult)))
     }
 
-    parseResult match {
-      case Some((parseResult, usedIdentNodes)) =>
+    parseResult.map {
+      case (parseResult, usedIdentNodes) =>
         val (result, duration) = {
-          TimeUtils.time(generateCpg(parseResult, diffGraph, usedIdentNodes))
+          TimeUtils.time(generateCpg(parseResult, localDiff, usedIdentNodes))
         }
         val path = parseResult.jsSource.originalFilePath
         result match {
           case Failure(exception) =>
             logger.warn(s"Failed to generate CPG for '$path'!", exception)
-            Iterator.empty
-          case Success(cpg) =>
+          case _ =>
             logger.info(s"Processed file '$path'")
             report.updateReportDuration(path, duration)
-            Iterator(cpg)
+            diffGraph.moveFrom(localDiff)
         }
-      case None =>
-        Iterator.empty
     }
   }
 
   private def generateCpg(parseResult: ParseResult,
                           diffGraph: DiffGraph.Builder,
-                          usedIdentNodes: Set[String]): Try[DiffGraph] = {
+                          usedIdentNodes: Set[String]): Try[Unit] = {
```

Comment:
`Try[Unit]` is a bit obscure, can't we return the DiffGraph here, i.e. without calling `.build()` on it, which seems to be the main change?